pkg/postgres/tests/conn_test.go

```diff
@@ -57,21 +57,29 @@ func (s *postgresConnTestSuite) TestConnBegin() {
 	s.mockPgxPoolConn.EXPECT().Begin(gomock.Any()).Times(1).DoAndReturn(func(ctx context.Context) (pgx.Tx, error) {
 		return s.mockTx, nil
 	})
-	tx1, err := s.conn.Begin(context.Background())
+	tx1, ctx1, err := s.conn.Begin(context.Background())
 	s.NoError(err)
 	s.NotNil(tx1)
+	fromContext, ok := postgres.TxFromContext(ctx1)
+	s.Same(fromContext, tx1)
+	s.True(ok)
+
+	//// With tx
+	ctxWithTx := postgres.ContextWithTx(context.Background(), &postgres.Tx{})
+	s.PanicsWithValue("it is not allowed to wrap a tx twice", func() {
```

Comment:
Is this limitation new or has it always existed?  Really the executing transaction shouldn't know or care how many transactions deep it is.  It should just do its job in relation to its transaction.

Reply:
It was there added in https://github.com/stackrox/stackrox/pull/5868
https://github.com/stackrox/stackrox/blob/67a66dc798a6ef6a2f2d6d392a05c6fb458d567c/pkg/postgres/tx.go#L104-L105